src/Hosting/Hosting/src/Internal/HostedServiceExecutor.cs

```diff
@@ -2,53 +2,39 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using Microsoft.Extensions.Hosting;
-using Microsoft.Extensions.Logging;
 
 namespace Microsoft.AspNetCore.Hosting;
 
 internal sealed class HostedServiceExecutor
 {
     private readonly IEnumerable<IHostedService> _services;
-    private readonly ILogger<HostedServiceExecutor> _logger;
 
-    public HostedServiceExecutor(ILogger<HostedServiceExecutor> logger, IEnumerable<IHostedService> services)
+    public HostedServiceExecutor(IEnumerable<IHostedService> services)
     {
-        _logger = logger;
         _services = services;
     }
 
-    public Task StartAsync(CancellationToken token)
+    public async Task StartAsync(CancellationToken token)
     {
-        return ExecuteAsync(service => service.StartAsync(token));
-    }
-
-    public Task StopAsync(CancellationToken token)
-    {
-        return ExecuteAsync(service => service.StopAsync(token), throwOnFirstFailure: false);
+        foreach (var service in _services)
```

Comment:
@Tratcher Is it important that these happen sequentially?  (vs `WhenAll`)

Reply:
This is only used in the legacy WebHost, I wouldn't change the behavior. I wouldn't want to introduce race conditions for components that were interdependnet.